packages/core/src/stepFunctionsWorkflowStudio/WorkflowStudioEditor.ts

```diff
@@ -0,0 +1,180 @@
+    public defaultTemplateName: string
+    // TODO: add fileStates and autoSaveFileStates variables to store the state of the file and handle auto-save
+    private getWebviewContent: () => Promise<string>
+
+    public constructor(
+        textDocument: vscode.TextDocument,
+        webviewPanel: vscode.WebviewPanel,
+        context: vscode.ExtensionContext,
+        fileId: string,
+        getWebviewContent: () => Promise<string>
+    ) {
+        this.getWebviewContent = getWebviewContent
+        this.documentUri = textDocument.uri
+        this.webviewPanel = webviewPanel
+        this.workSpacePath = path.dirname(textDocument.uri.fsPath)
+        this.defaultTemplatePath = textDocument.uri.fsPath
+        this.defaultTemplateName = path.basename(this.defaultTemplatePath)
+        this.fileId = fileId
+
+        telemetry.workflowStudio_opened.record({
+            id: this.fileId,
+        })
+
+        this.setupWebviewPanel(textDocument, context)
+    }
+
+    public get onVisualizationDisposeEvent(): vscode.Event<void> {
+        return this.onVisualizationDisposeEmitter.event
+    }
+    public getPanel(): vscode.WebviewPanel | undefined {
+        if (!this.isPanelDisposed) {
+            return this.webviewPanel
+        }
+    }
+
+    public showPanel(): void {
+        this.getPanel()?.reveal()
+    }
+
+    public async refreshPanel(context: vscode.ExtensionContext) {
+        if (!this.isPanelDisposed) {
+            this.webviewPanel.dispose()
+            const document = await vscode.workspace.openTextDocument(this.documentUri)
+            this.setupWebviewPanel(document, context)
+        }
+    }
+
+    protected getText(textDocument: vscode.TextDocument): string {
+        return textDocument.getText()
+    }
+
+    /**
+     * Sets up the webview panel for Workflow Studio Editor. This includes creating the
+     * panel, setting up the webview content, and handling the communication between the webview
+     * and the extension context.
+     * @param textDocument The text document to be displayed in the webview panel.
+     * @param context The extension context.
+     * @private
+     */
+    private setupWebviewPanel(textDocument: vscode.TextDocument, context: vscode.ExtensionContext) {
+        const documentUri = textDocument.uri
+
+        const contextObject: WebviewContext = {
+            panel: this.webviewPanel,
+            textDocument: textDocument,
+            disposables: this.disposables,
+            workSpacePath: this.workSpacePath,
+            defaultTemplatePath: this.defaultTemplatePath,
+            defaultTemplateName: this.defaultTemplateName,
+            loaderNotification: undefined,
+            fileId: this.fileId,
+        }
+
+        async function cancelOpenInWorkflowStudio(reason: string) {
+            contextObject.panel.dispose()
+            await vscode.commands.executeCommand('vscode.openWith', documentUri, 'default')
+
+            telemetry.workflowStudio_closed.emit({
```

Comment:
Is knowing that people clicked the "closed" button worth gathering?

Reply:
I took a look again and agree that in this case collecting it (the number of times the tab is closed) does not provide much value, I think it makes sense to remove it from here.
At the same time, keeping the same metric for the case when the user cancels opening the integration does make sense in my opinion, this should help us calculate the adoption rate